src/Kestrel/Internal/DefaultHttpsProvider.cs

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Server.Kestrel.Core;
+using Microsoft.AspNetCore.Server.Kestrel.Core.Internal;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Internal
+{
+    public class DefaultHttpsProvider : IDefaultHttpsProvider
+    {
+        private const string AspNetHttpsOid = "1.3.6.1.4.1.311.84.1.1";
+
+        public Func<X509Certificate2> DefaultCertificateResolver { get; set; } = FindDevelopmentCertificate;
+
+        public void ConfigureHttps(ListenOptions listenOptions)
+        {
+            listenOptions.UseHttps(DefaultCertificateResolver());
+        }
+
+        private static X509Certificate2 FindDevelopmentCertificate()
+        {
+            // TODO: replace this with call to
+            // CertificateManager.FindCertificates(CertificatePurpose.HTTPS, StoreName.My, StoreLocation.CurrentUser, isValid: true)
+            // when that becomes available.
+            using (var store = new X509Store(StoreName.My, StoreLocation.CurrentUser))
```

Comment:
Weird, I was sure I had pushed the change that removed all this logic and just called [`CertificateManager.ListCertificate`](https://github.com/aspnet/Common/blob/9154a85af62b4e461733f32e062498140ecd7cad/shared/Microsoft.AspNetCore.Certificates.Generation.Sources/CertificateManager.cs#L42). You can just call that like this:

```csharp
CertificateManager.ListCertificates(CertificatePurpose.HTTPS, StoreName.My, StoreLocation.CurrentUser, isValid: true, requireExportable: false);
```

Reply:
It's possible something blew up during a rebase. I'll bring it back in :).